### ISSUE TITLE
Get the same reward for all input!

### ISSUE THREAD

**Issue by SongyiGao** [NVIDIA-AI-IOT/torch2trt#137]
I convert pretrained  model to tensorrt, I print out the result, finding that the result of model_trt is always the same whatever the input is.
`Segmentator(
  (backbone): Backbone(
    (conv1): Conv2d(5, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
    (bn1): BatchNorm2d(32, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
    (relu1): LeakyReLU(negative_slope=0.1)
    (enc1): Sequential(
      (conv): Conv2d(32, 64, kernel_size=(3, 3), stride=[1, 2], padding=(1, 1), bias=False)
      (bn): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual_0): BasicBlock(
        (conv1): Conv2d(64, 32, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(32, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(32, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (enc2): Sequential(
      (conv): Conv2d(64, 128, kernel_size=(3, 3), stride=[1, 2], padding=(1, 1), bias=False)
      (bn): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual_0): BasicBlock(
        (conv1): Conv2d(128, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (enc3): Sequential(
      (conv): Conv2d(128, 256, kernel_size=(3, 3), stride=[1, 2], padding=(1, 1), bias=False)
      (bn): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual_0): BasicBlock(
        (conv1): Conv2d(256, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
      (residual_1): BasicBlock(
        (conv1): Conv2d(256, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (enc4): Sequential(
      (conv): Conv2d(256, 512, kernel_size=(3, 3), stride=[1, 2], padding=(1, 1), bias=False)
      (bn): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual_0): BasicBlock(
        (conv1): Conv2d(512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(256, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
      (residual_1): BasicBlock(
        (conv1): Conv2d(512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(256, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (enc5): Sequential(
      (conv): Conv2d(512, 1024, kernel_size=(3, 3), stride=[1, 2], padding=(1, 1), bias=False)
      (bn): BatchNorm2d(1024, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual_0): BasicBlock(
        (conv1): Conv2d(1024, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(512, 1024, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(1024, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dropout): Dropout2d(p=0.01)
  )
  (decoder): Decoder(
    (dec5): Sequential(
      (upconv): ConvTranspose2d(1024, 512, kernel_size=[1, 4], stride=[1, 2], padding=[0, 1])
      (bn): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual): BasicBlock(
        (conv1): Conv2d(512, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(1024, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(1024, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dec4): Sequential(
      (upconv): ConvTranspose2d(512, 256, kernel_size=[1, 4], stride=[1, 2], padding=[0, 1])
      (bn): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual): BasicBlock(
        (conv1): Conv2d(256, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(512, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(512, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dec3): Sequential(
      (upconv): ConvTranspose2d(256, 128, kernel_size=[1, 4], stride=[1, 2], padding=[0, 1])
      (bn): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual): BasicBlock(
        (conv1): Conv2d(128, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(256, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(256, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dec2): Sequential(
      (upconv): ConvTranspose2d(128, 64, kernel_size=[1, 4], stride=[1, 2], padding=[0, 1])
      (bn): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual): BasicBlock(
        (conv1): Conv2d(64, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(128, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(128, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dec1): Sequential(
      (upconv): ConvTranspose2d(64, 32, kernel_size=[1, 4], stride=[1, 2], padding=[0, 1])
      (bn): BatchNorm2d(32, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
      (relu): LeakyReLU(negative_slope=0.1)
      (residual): BasicBlock(
        (conv1): Conv2d(32, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu1): LeakyReLU(negative_slope=0.1)
        (conv2): Conv2d(64, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(32, eps=1e-05, momentum=0.01, affine=True, track_running_stats=True)
        (relu2): LeakyReLU(negative_slope=0.1)
      )
    )
    (dropout): Dropout2d(p=0.01)
  )
  (head): Sequential(
    (0): Dropout2d(p=0.01)
    (1): Conv2d(32, 20, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  )
)
`

**Comment by jaybdub** [comment]
Hi SongyiGao,

Thanks for reaching out!  

It's likely that your model had an unsupported layer, but the conversion silently succeeded because a constant was added somewhere along the lines.

I've added a feature to master that will now warn for known torch methods that are unsupported.

Do you mind pulling the master branch and attempting to convert your model again?   It should report the unsupported layers.

Please let me know if this works for you.

Best,
John
